framework/include/utils/InputParameters.h

```diff
@@ -841,60 +850,47 @@ class InputParameters : public Parameters
    */
   std::set<std::string> reservedValues(const std::string & name) const;
 
-  ///@{
   /**
-   * Get/set a string representing the location (i.e. filename,linenum) in the input text for the
+   * @return A string representing the location (i.e. filename,linenum) in the input text for the
    * block containing parameters for this object.
    */
-  std::string & blockLocation() { return _block_location; }
-  const std::string & blockLocation() const { return _block_location; }
-  ///@}
+  std::string blockLocation() const;
 
-  ///@{
   /**
-   * Get/set a string representing the full HIT parameter path from the input file (e.g.
+   * @return A string representing the full HIT parameter path from the input file (e.g.
    * "Mesh/foo") for the block containing parameters for this object.
    */
-  std::string & blockFullpath() { return _block_fullpath; }
-  const std::string & blockFullpath() const { return _block_fullpath; }
-  ///@}
+  std::string blockFullpath() const;
 
-  ///@{
   /**
-   * Get/set a string representing the location in the input text the parameter originated from
-   * (i.e. filename,linenum) for the given param.
+   * @return The hit node associated with setting the parameter \p param, if any
    */
-  const std::string & inputLocation(const std::string & param) const
-  {
-    return at(param)._input_location;
-  }
-  std::string & inputLocation(const std::string & param) { return at(param)._input_location; }
-  ///@}
+  const hit::Node * getHitNode(const std::string & param) const;
+  /**
+   * Sets the hit node associated with the parameter \p param to \p node
+   */
+  void setHitNode(const std::string & param, const hit::Node & node);
```

Comment:
there is no value in having a getter setter pair like this. My suggestion would be to make setHitNode private and add Builder, *Factory as friends (or use the pass-key idiom).

Reply:
Done. Good call